### DISCUSSION TITLE
Is managed sieve working?

### DISCUSSION THREAD

**Discussion by jessec** [docker-mailserver/docker-mailserver#3653] [Q&A] [answered]
```
0.0.0.0:25->25/tcp, :::25->25/tcp, 0.0.0.0:143->143/tcp, :::143->143/tcp, 0.0.0.0:465->465/tcp, :::465->465/tcp, 0.0.0.0:587->587/tcp, :::587->587/tcp, 110/tcp, 995/tcp, 0.0.0.0:993->993/tcp, :::993->993/tcp**, 4190/tcp**
```

I can't seem to get managed sieve working. I see only 1 port is it possible the sieve deamon is not running?

**Comment by polarathene** [comment]
You should provide more context (_`compose.yaml` + other config to reproduce_), have you tried `:edge` tag? There was sieve fixes added there that will be part of the soon to release v13.

**Comment by jessec** [comment]
I followed the documentation and I opened the 4190 port. Is there a way to identify if the sieve plugin is loaded and working? Maybe with a `ps | grep "sieve"` ???

I also can't telnet into the port:

```console
$ telnet  15.45.219.48 4190
telnet: connect to address 15.45.219.48:  Connection refused
```

and inside the DMS container I see this: 

```
root@lopkyhj:/etc/dovecot# ls /etc/dovecot/protocols.d/   
imapd.protocol	lmtpd.protocol	managesieved.protocol.disab  pop3d.protocol.disab
```

> `managesieved.protocol.disab`

is it still disabled?

# `compose.yaml`

```yml
ports:
  - "4190:4190"
environment:
  - ENABLE_MANAGESIEVE=1
```

**Reply by polarathene** [reply, marked as answer]
> and inside the docker mailserver container I see this:
> 
> ```
> root@lopkyhj:/etc/dovecot# ls /etc/dovecot/protocols.d/
> imapd.protocol lmtpd.protocol managesieved.protocol.disab pop3d.protocol.disab
> ```
> 
> > `managesieved.protocol.disab`
> is it still disabled?

It doesn't seem like the ENV logic ran for you?

https://github.com/docker-mailserver/docker-mailserver/blob/2a716cf4a4e0eec9ec12d99fcc6e9c5e625a63f8/target/scripts/startup/setup.d/dovecot.sh#L52-L57

That should have renamed it to enable it.

We have an [incredibly basic test](https://github.com/docker-mailserver/docker-mailserver/blob/2a716cf4a4e0eec9ec12d99fcc6e9c5e625a63f8/test/tests/parallel/set1/dovecot/dovecot_sieve.bats#L55-L60) for the [Manage Sieve](https://docker-mailserver.github.io/docker-mailserver/latest/config/advanced/mail-sieve/#manage-sieve) feature, but it does check that it is running AFAIK. Thus it should have caught any problem you're experiencing and I think this is a configuration issue on your end?

---

Please make sure your config changes are applied with a fresh container instance, not a restarted/resumed one. Try with `docker compose up --force-recreate`.

If this still doesn't resolve the issue, you should raise a bug report. I'm presently short of time to reproduce and investigate if there is an actual bug on our end.

**Reply by jessec** [reply]
Yes I had to recreate that was the problem thanks!
